crates/common/tedge_utils/src/file.rs

```diff
@@ -115,22 +124,32 @@ pub async fn move_file(
         false => None,
     };
 
-    tokio::fs::rename(src_path, dest_path).await?;
-    debug!("Moved file from {:?} to {:?}", src_path, dest_path);
-
-    let file_permissions = if let Some(mode) = original_permission_mode {
-        // Use the same file permission as the original one
-        PermissionEntry::new(None, None, Some(mode))
-    } else {
-        // Set the user, group, and mode as given for a new file
-        new_file_permissions
-    };
+    // Copy source to destination using rename. If that one fails due to cross-filesystem, use copy and delete.
+    // As a ErrorKind::CrossesDevices is nightly feature we call copy and delete no matter what kind of error we get.
+    match tokio::fs::rename(src_path, dest_path).await {
+        Ok(_) => {
+            let file_permissions = if let Some(mode) = original_permission_mode {
+                // Use the same file permission as the original one
+                PermissionEntry::new(None, None, Some(mode))
+            } else {
+                // Set the user, group, and mode as given for a new file
+                new_file_permissions
+            };
+
+            file_permissions.apply(dest_path)?;
+            debug!(
+                "Applied permissions: {:?} to {:?}",
+                file_permissions, dest_path
+            );
```

Comment:
This must also be done in the copy-and-delete case.

Reply:
Agree with @didier-wenzek